docs/2-building-blocks-of-oop-part-1/hometask/src/Point.ts

```diff
@@ -0,0 +1,36 @@
+export class Point {
+  private x: number;
+  private y: number;
+
+  constructor();
+  constructor(x: number, y: number);
+  constructor(x?: number, y?: number) {
+    this.x = x || 0;
+    this.y = y || 0;
+  }
+
+  toString(): string {
+    return `(${this.x}, ${this.y})`;
+  }
+
+  private calculatePointsDistance(passedX: number = 0, passedY: number = 0): number {
```

Comment:
try to remove this method.

The logick can be placed inside `distance` method, but code duplication should be avoided.